logging/DetailedErrorFormatter.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace SilverStripe\Framework\Logging;
+
+use Monolog\Logger;
+use Monolog\Handler\AbstractProcessingHandler;
+use Monolog\Formatter\FormatterInterface;
+
+/**
+ * Monolog-compatible error handler that will output a detailed error message to the screen.
+ */
+class DetailedErrorFormatter implements FormatterInterface
+{
+	public function format(array $record)
+	{
+		if(isset($record['context']['exception'])) {
+			$exception = $record['context']['exception'];
+			$context = array(
+				'code' => $exception->getCode(),
+				'message' => 'Uncaught ' . get_class($exception) . ': ' . $exception->getMessage(),
+				'file' => $exception->getFile(),
+				'line' => $exception->getLine(),
+				'trace' => $exception->getTrace(),
+			);
+		} else {
+			$context = $record['context'];
+			foreach(array('code','message','file','line') as $key) {
+				if(!isset($context[$key])) {
+					$context[$key] = isset($record[$key]) ? $record[$key] : null;
+				}
+			}
+
+			$trace = debug_backtrace();
+
+			// Filter out monolog plumbing from the trace
+			$i = $this->findInTrace($trace, $context['file'], $context['line']);
+			if($i !== null) {
+				$trace = array_slice($trace, $i);
+			}
+
+			$context['trace'] = $trace;
+		}
+
+		return $this->output(
+			$context['code'],
+			$context['message'],
+			$context['file'],
+			$context['line'],
+			$context['trace']
+		);
+	}
+
+	public function formatBatch(array $records) {
+		return implode("\n", array_map(array($this, 'format'), $records));
+	}
+
+	/**
+	 * Find a call on the given file & line in the trace
+	 * @param array $trace The result of debug_backtrace()
+	 * @param string $file The filename to look for
+	 * @param string $line The line number to look for
+	 * @return int|null The matching row number, if found, or null if not found
+	 */
+	protected function findInTrace(array $trace, $file, $line) {
+		foreach($trace as $i => $call) {
+			if(isset($call['file']) && isset($call['line']) && $call['file'] == $file && $call['line'] == $line) {
+				return $i;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Render a developer facing error page, showing the stack trace and details
+	 * of the code where the error occured.
+	 *
+	 * @param unknown_type $errno
```

Comment:
I think @assertchris would be angry at me if I didn't ask you to type these. :) `mixed` is probably ok, but I'd guess int, string, string, int, array